test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/ExportsTest.scala

```diff
@@ -966,6 +968,80 @@ object ExportsTest extends JasmineTest {
       expect(jsPackage.ExportObjSetterNamed_=().x).toBe(1)
     }
 
+    it("should export js.Function values of new js.Object{...} - #1899") {
+
+      // Test that the bug is fixed for js.Any classes.
+
+      def getObj(): js.Object = new js.Object {
```

Comment:
It would also be good to test with an anonymous subclass of a strict subclass of `js.Object`.